tests/ui/merge.spec.js

```diff
@@ -0,0 +1,86 @@
+"use strict";
+
+var Stubs = require("./stubs");
+var stubdata = require("./stubdata");
+
+var webdriver = require("selenium-webdriver");
+var By = require("selenium-webdriver").By;
+var until = require("selenium-webdriver").until;
+var test = require("selenium-webdriver/testing");
+var assert = require("selenium-webdriver/testing/assert");
+
+test.describe("Merge page", function() {
+    this.timeout(5000);
+
+    var driver;
+    var stubs;
+
+    test.beforeEach(() => {
+        stubs = new Stubs();
+        driver = new webdriver.Builder()
+            .forBrowser("chrome")
+            .build();
+        driver.manage().timeouts().implicitlyWait(200)
+            .then(() => stubs.init(driver));
+    });
+
+    test.afterEach((done) => driver.quit().then(done));
+
+    test.describe("squash merging rewriting history", () => {
+        test.it("succeeds", () => {
+            stubs.stub(stubdata.MERGABLE_PR);
+
+            tryRewriteMerge("someone/some-repo/1", "PR title");
+            stubs.queueSuccessfulRewriteHistory();
+            driver.wait(until.elementLocated(By.className("PullRequestOverview_state_merged")), 500);
+            driver.findElement(By.className("PullRequestOverview_state_merged")).getText()
+                .then((text) => assert(text).equalTo("Merged"));
+        });
+
+        test.it("handles errors", () => {
+            stubs.stub(stubdata.MERGABLE_PR);
+
+            tryRewriteMerge("someone/some-repo/1", "PR title");
+            stubs.queueUnsuccessfulRewriteHistory();
+            driver.wait(until.elementLocated(By.className("MergeButton_error")), 500);
+            driver.findElement(By.className("MergeButton_error")).getText()
```

Comment:
Yeah, I'll totally do this if you're happy with the general structure and set-up :)